File: 0539/Solution.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        int i = 0, n = timePoints.size(), res = Integer.MAX_VALUE;
        int[] min = new int[n];
        for(String s : timePoints){
            int h = Integer.parseInt(s.substring(0, 2));
            int m = Integer.parseInt(s.substring(3));
            min[i++] = h * 60 + m;
        }
        Arrays.sort(min);
        for(i = 1; i < n; i++){
            res = Math.min(res, min[i] - min[i-1]);
        }
        return Math.min(res, 24 * 60 + min[0] -  min[n - 1]);
    }
}

File: 0539/Solution.go
func findMinDifference(timePoints []string) int {
    res := math.MaxInt32
    min := make([]int, len(timePoints))
    for i, s := range timePoints {
        h, _ := strconv.Atoi(s[0:2])
        m, _ := strconv.Atoi(s[3:])
        min[i] = 60 * h + m
    }
    sort.Ints(min)
    for i := 1; i < len(min); i++ {
        cur := min[i] - min[i-1]
        if res > cur {
            res = cur
        }
    }
    rev := 24 * 60 + min[0] - min[len(min) - 1]
    if rev < res {
        return rev
    } 
    return res
}
